### PR TITLE
Read the TTL from the configuration service

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
- The `eventually` now wraps only the attempt to fetch the ledger identifier and happens only the first time the test tool tries to get in touch with the participant - this still ensures that the tool can wait for slow participants but won't recover in case of ledgers with an intermittent connection to the tool (which was a non-issue in the first place)
- The ledger identifier is retrieved only once when the connection to the participant is first established - this makes it less resilient in the face of changing ledger identifiers, which would nonetheless have been an issue anyhow due to the way in which the tool isolates the tests
- The TTL is retrieved from the ledger configuration service as the maximum possible value
- The existing CLI option can be used to adjust the TTL _down_
- No matter what, the TTL will _always_ be clipped between the maximum and minimum values returned by the ledger configuration service
- The ledger configuration service is contacted only once upon connection to the participant, so changing configuration will not be picked up along the way
- In general, tests that can change the global state of the ledger (reset, time, configuration, id are the first to come to mind) _should_ be run separately so to not interfere with other tests

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantSession.scala [hunk 1]
```diff
@@ -22,29 +27,61 @@ private[participant] final class ParticipantSession(
...
+    ledgerIdF
+      .flatMap { id =>
+        SingleItemObserver
+          .first[GetLedgerConfigurationResponse](services.configuration
+            .getLedgerConfiguration(new GetLedgerConfigurationRequest(id), _))
+          .map(_.get.getLedgerConfiguration)
+      }
+      .map { configuration =>
+        val factor = config.commandTtlFactor
+        val min = (configuration.getMinTtl.seconds * 1E9) + configuration.getMinTtl.nanos.toDouble
```
Comment (gerolf-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantSession.scala]: `1E9` <- what a rare sight to behold :)

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantSession.scala [hunk 2]
```diff
@@ -22,29 +27,61 @@ private[participant] final class ParticipantSession(
...
+    }
+
+  // The time-to-live for commands defaults to the maximum value as defined by the ledger
+  // configuration and can be adjusted down. Regardless the output of the adjustment, the
+  // value is always going to be clipped by the minimum and maximum configured values.
+  // Note that this value is going to never change after it's read the first time, so changing
+  // ledger configuration across tests may have wildly unexpected consequences. In general the
+  // test tool is designed to have tests work in isolation and tests addressing changes to the
+  // global state of the ledger should be isolated in their own test runs.
+  private[this] val ttlNanosF: Future[Long] =
```
Comment (gerolf-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantSession.scala]: Would it make sense to make this a `scala.concurrent.duration.Duration`? Using duration is much safer than passing around a long that hopefully never changes its meaning.

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantSession.scala (hunk 1 above)
Comment (gerolf-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantSession.scala]: Alternatively:
```
import scala.concurrent.duration._

val min = (configuration.getMinTtl.seconds.seconds + configuration.getMinTtl.nanos.nanos
// ...
// multiplying by factor and the min max functions would work just the same
```

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantTestContext.scala [hunk 3]
```diff
@@ -96,7 +94,7 @@ private[testtool] final class ParticipantTestContext private[participant] (
     val identifierSuffix: String,
     referenceOffset: LedgerOffset,
     services: LedgerServices,
-    commandTtlFactor: Double)(implicit ec: ExecutionContext) {
+    ttlNanos: Long)(implicit ec: ExecutionContext) {
```
Comment (gerolf-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantTestContext.scala]: Preferably we should use `Duration` here.

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantSession.scala (hunk 2 above)
Comment (stefanobaghino-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantSession.scala]: Makes a lot of sense, thanks, will do.

Comment (stefanobaghino-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantSession.scala]: Addressed by https://github.com/digital-asset/daml/pull/3052/commits/f28736e5df07b0d30fd1b1d174e63fa974c75fde

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantTestContext.scala (hunk 3 above)
Comment (stefanobaghino-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantTestContext.scala]: Addressed by https://github.com/digital-asset/daml/pull/3052/commits/f28736e5df07b0d30fd1b1d174e63fa974c75fde

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantSession.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantSession.scala]: Addressed by https://github.com/digital-asset/daml/pull/3052/commits/f28736e5df07b0d30fd1b1d174e63fa974c75fde

**Comment by stefanobaghino-da** [comment]
@gerolf-da Your concerns should have been addressed by https://github.com/digital-asset/daml/pull/3052/commits/f28736e5df07b0d30fd1b1d174e63fa974c75fde, on top of that I also made sure to update documentation as part of https://github.com/digital-asset/daml/pull/3052/commits/8efcdd4cb8c2327da1e5fa8b20b81aaec11bf80c

**Pull request merged**